// vidi/mongo/user.js
var mongoose = require('mongoose'),
	model = require('./model'),
	video = require('./video'),
	crypto = require('../crypto');


exports.userId = null;

exports.signup = function(user,socket){
	exports.createUser(user,socket);
}


exports.createUser = function(User,socket){
	model.user.findOne({username:User.username}, function (err, user) {
  		if (err) { console.log(err); }
  		//IF USERNAME UNKNOWN
	  	if(user==undefined){

	  		//CREATE USER
			var user = new model.user({ 
				username : User.username,
				password:crypto.encrypt(User.password),
				video:{
					liked:0,
					shared:0,
					comments:0
				}
			});
			console.log(user);

			//SAVE USER
			user.save(function (err) {
			  if (err) { 
			  	exports.createUser.fail(socket,'Server fail :(');
			  	console.log(err); 
			  }
			  console.log(user.username + ' ajouté avec succès !');
			  exports.createUser.success(socket,'Sign Up :)',user);

			});

		}else exports.createUser.fail(socket,'Username already used');
	});
}

exports.createUser.fail = function(socket,alert){
	console.log('createUser fail');
	socket.emit('form-alert',alert,'error');
}
exports.createUser.success = function(socket,alert,user){
	console.log('createUser success');
	exports.id = user._id;
	socket.emit('form-alert',alert,'success',user);
}

////////////////////////////////////////////////////////

exports.login = function(user,socket){
	exports.checkUser(user,socket);
}

exports.checkUser = function(User,socket){
	model.user.findOne({username:User.username}, function (err, user) {
	  if (err) { console.log(err); }
	  console.log(user)
	  if(user!=undefined){
	  	
	  	if(crypto.decrypt(user.password) == User.password)
	  		exports.checkUser.success(socket,'Success !',user);
		else exports.checkUser.fail(socket,'wrong password');

	  }else exports.checkUser.fail(socket,'wrong username & password');

	});
}

exports.checkUser.fail = function(socket,alert){
	console.log('check fail');
	socket.emit('form-alert',alert,'error');
}

exports.checkUser.success = function(socket,alert,user){	
	
	exports.id = user._id;
	socket.emit('form-alert',alert,'success',user);
	video.check(user._id,socket);

	console.log('check success',user._id);
}
