// vidi/mongo/video.js
var mongoose = require('mongoose'),
		model = require('./model'),
		fs = require('fs'),
		formidable = require('formidable');

exports.upload = function(req,res,io,dir,_userId){

	var form = new formidable.IncomingForm();

	var socket = io.sockets.in(req.sessionID);

	socket.emit('upload-start');

	var _ratio = 0;
	form.addListener('progress', function(bytesReceived, bytesExpected){
    
    var ratio = Math.round((bytesReceived/bytesExpected)*100);
    if(ratio>=(_ratio+5)){
    	_ratio = ratio;
    	socket.emit('upload-progress',ratio);
    }
    if(ratio==100){
    	console.log('upload finish');
    }

  });


  form.uploadDir = dir + '/data/';
  form.on('file', function(field, file) {

    socket.emit('upload-finish');

    fs.rename(file.path, form.uploadDir + "/" + file.name);

    var video = new model.video({
			userId: _userId,
			source : form.uploadDir + "/" + file.name,
			title : file.name,
			desc : '',  
		  liked: 0,
		  comments : [],
		});

		video.save(function (err) {
		  if (err) { 
		  	exports.upload.fail(socket,'Server upload fail :(');
		  	console.log(err); 
		  }
		  console.log(video.title + ' ajouté avec succès !');

		 
		  model.user.findOne({_id:_userId},function(err,user){
		  	if(err)console.log('user shared inc error',err);
		  	user.video.shared++;
		  	var User=user;
		  	user.save();
		  	exports.upload.success(socket,'Server upload success :)',User);
		  });


		});

  });

	form.parse(req, function(err, fields, files) {
    res.writeHead(200);
    res.end();
  });

  console.log(form);
}

exports.upload.fail = function(socket,msg,user){

}

exports.upload.success = function(socket,msg,user){
	socket.emit('update-profil',msg,user);
}

exports.check = function(id,socket){
	model.video.find({userId:id},function(err,videos){
		if(err)console.log('get-videos error',err);

		console.log(videos);
		socket.emit('get-user-videos',videos);
	});
}
